dist/extras/amd.js

```diff
@@ -3,132 +3,178 @@
-    for (let i = 0; i < amdDefineDeps.length; i++) {
-      const id = amdDefineDeps[i];
+
+    for (let i = 0; i < dependencies.length; i++) {
+      const id = dependencies[i];
       const index = setters.length;
+
       if (id === 'require') {
         depModules[i] = unsupportedRequire;
         splice++;
-      }
-      else if (id === 'module') {
+      } else if (id === 'module') {
         depModules[i] = module;
         splice++;
-      }
-      else if (id === 'exports') {
+      } else if (id === 'exports') {
         depModules[i] = exports;
         splice++;
-      }
-      else {
-        // needed for ie11 lack of iteration scope
+      } else {
         const idx = i;
         setters.push(function (ns) {
-          depModules[idx] = ns.__useDefault ? ns.default : ns;
+          depModules[idx] = ns;
         });
       }
-      if (splice)
-        amdDefineDeps[index] = id;
+      if (splice) {
+        dependencies[index] = id;
+      }
     }
-    if (splice)
-      amdDefineDeps.length -= splice;
-    const amdExec = amdDefineExec;
-    return [amdDefineDeps, function (_export) {
-      _export({ default: exports, __useDefault: true });
-      return {
-        setters: setters,
-        execute: function () {
-          module.exports = amdExec.apply(exports, depModules) || module.exports;
-          if (exports !== module.exports)
-            _export('default', module.exports);
+    if (splice) {
+      dependencies.length -= splice;
+    }
+	  if(dependencies.length > 0) {
+      for(let j = 0; j < dependencies.length; j++) {
+        if(dependencies[j].substr(dependencies[j].length-1) 
+          && dependencies[j].indexOf("/") >= 0) {
+          dependencies[j] = dependencies[j]+".js";
         }
-      };
-    }];
-  }
+      }
+    }
 
-  // hook System.register to know the last declaration binding
-  let lastRegisterDeclare;
-  const systemRegister = systemPrototype.register;
-  // if we have named register support continue to use it
-  if (systemRegister.length === 3) {
-    systemPrototype.register = function (name, deps, declare) {
-      if (typeof name !== 'string')
-        lastRegisterDeclare = deps;
-      systemRegister.apply(this, arguments);
-    };
+    return [dependencies, depModules, setters];
   }
-  else {
-    systemPrototype.register = function (deps, declare) {
-      lastRegisterDeclare = declare;
-      systemRegister.apply(this, arguments);
-    };
-  }
-
-  const getRegister = systemPrototype.getRegister;
-  systemPrototype.getRegister = function () {
-    const register = getRegister.call(this);
-    // if its an actual System.register leave it
-    if (register && register[1] === lastRegisterDeclare)
-      return register;
 
-    // otherwise AMD takes priority
-    // no registration -> attempt AMD detection
-    if (!amdDefineDeps)
-      return register || emptyInstantiation;
+  /**
+   * Hook System.register
+   */
+  (function hookSystemRegister(lastRegisterDeclare) {
```

Comment:
Encapsulated the hooking behaviour with a self-calling function to keep it in its own scope